apiserver/client/status.go

```diff
@@ -522,6 +522,23 @@ func makeMachineStatus(machine *state.Machine) (status params.MachineStatus) {
 			logger.Debugf("error fetching public address: %q", err)
 		}
 		status.DNSName = addr.Value
+
+		mAddrs := machine.Addresses()
+		if len(mAddrs) == 0 {
+			logger.Debugf("no IP addresses fetched for machine %q", instid)
+			// At least give it the newly created DNSName address, if it exists.
+			if addr.Value != "" {
+				mAddrs = append(mAddrs, addr)
+			}
+		}
+		for _, mAddr := range mAddrs {
```

Comment:
This loop is in quite similar to what network.FilterUnusableHostPorts() does, unfortunately not for []network.Address.
Nevertheless, please use the same approach here - switch on mAddr.Scope and filter ScopeMachineLocal and ScopeLinkLocal. This will cover the full loopback ranges 127/8 and ::1/128 on both IPv4 and IPv6 addresses, and will also not show the linkewise-unusable link local addresses.